packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -3043,22 +3043,43 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin {
     // If text is obscured, the entire sentence should be treated as one word.
     if (obscureText) {
       return TextSelection(baseOffset: 0, extentOffset: _plainText.length);
-    // If the word is a space, on iOS try to select the previous word instead.
-    // On Android try to select the previous word instead only if the text is read only.
+    // On iOS, select the previous word if there is a previous word, or select
+    // the the whitespace to the next word if there is a next word. Select
+    // nothing if there is neither a previous word nor a next word.
+    //
+    // If the platform is Android and the text is read only, try to select the
+    // previous word if there is one; otherwise, select the whitespace at the
+    // position.
```

Comment:
```diff
- the whitespace
+ the single whitespace
```

Just to clarify that it won't select a bunch of whitespace, just one space character (right?).